.github/workflows/build.yaml

```diff
@@ -18,38 +34,86 @@ jobs:
+      if: github.ref == 'refs/heads/main' && matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON
+      run: |
+        # Set up the GitHub user and email as author for the release commit.
+        git config --global user.name $USER_NAME
+        git config --global user.email $USER_EMAIL
+        git config --list --global
+        # First generate the CHANGELOG diff to use as release notes.
+        cz changelog --dry-run --incremental --unreleased-version "Release Notes" > dist/RELEASE_NOTES.md
+        # Update the CHANGELOG.md.
+        cz changelog
+        cz bump
     - name: Build the package
-      # We don't need to check and test the package separately
-      # because `make dist` runs those targets first and only
-      # builds the package if they succeed.
-      run: make dist
-    - name: Build the docs
-      run: make docs
+      # We don't need to check and test the package separately because `make dist` runs
+      # those targets first and only builds the package if they succeed.
+      run: |
+        make dist requirements sbom
+    - name: Compute package hash
+      if: matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON
+      id: compute-hash
+      shell: bash
+      run: |
+        set -euo pipefail
+        # Find the paths to the files that will be included in the release.
+        TARBALL_PATH=$(find dist -name "*.tar.gz")
+        WHEEL_PATH=$(find dist -name "*.whl")
+        SBOM_PATH=$(find dist -name "*-sbom.json")
+        HTML_DOCS_PATH=$(find dist -name *-docs-html.zip)
+        # Make sure dist/RELEASE_NOTES.md (which contains the release notes) exists.
+        touch dist/RELEASE_NOTES.md
+        NOTES_PATH=$(find dist -name RELEASE_NOTES.md)
+        # Compute the sha digest for all the release files and encode them using base64.
+        DIGEST=$(sha256sum $TARBALL_PATH $WHEEL_PATH $SBOM_PATH $HTML_DOCS_PATH $NOTES_PATH | base64 -w0)
+        echo "digest of artifacts is $DIGEST."
+        # Set the computed sha digest as the output of this job.
+        echo "::set-output name=artifacts-sha256::$DIGEST"
+    - name: Upload the package artifact
+      # For now only generate artifacts for the specified OS and Python version in env variables.
+      # Currently reusable workflows do not support setting strategy property from the caller workflow.
+      if: matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON
+      uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
+      with:
+        name: artifact-${{ matrix.os }}-${{ matrix.python }}
+        path: dist
+        if-no-files-found: error
+        retention-days: 1
+    - name: Update release tag
+      # Push the release tag only once when branch is main on the specified OS and Python version.
+      if: github.ref == 'refs/heads/main' && matrix.os == env.ARTIFACT_OS && matrix.python == env.ARTIFACT_PYTHON
+      run: |
+        git push
+        git push --tags
```

Comment:
Do we want to change `staging` in a separate PR?
```bash
git checkout staging
git rebase main
git push
```
This would hardwire the branch name `staging` into the process, though, and maybe we should make it configurable? Probably separate discussion and PR 🤓

Reply:
Yes, that should be done in a separate PR.

Reply:
Issue https://github.com/jenstroeger/python-package-template/issues/266.